test-infra/ansible/ci-runner/entrypoint.sh

```
@@ -0,0 +1,70 @@
+if [ -z "$GH_PAT" ] || [ -z "$GH_REPO_URL" ]; then
+    echo "Error: GH_PAT and GH_REPO_URL environment variables are required."
+    exit 1
+fi
+REPO_NAME=$(echo "${GH_REPO_URL}" | sed -e 's|https://github.com/||' -e 's|/$||')
+
+echo "--> Getting Registration Token for ${REPO_NAME}..."
+REG_TOKEN=$(curl -sX POST \
+  -H "Accept: application/vnd.github+json" \
+  -H "Authorization: Bearer ${GH_PAT}" \
+  -H "X-GitHub-Api-Version: 2022-11-28" \
+  https://api.github.com/repos/${REPO_NAME}/actions/runners/registration-token | jq .token --raw-output)
```

Shellcheck (0.11.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)